src/argilla_sdk/settings/_resource.py

```diff
@@ -257,9 +258,9 @@ def __repr__(self) -> str:
     #  Private methods  #
     #####################
 
-    def __fetch_fields(self) -> List[FieldType]:
+    def __fetch_fields(self) -> List[TextField]:
         models = self._client.api.fields.list(dataset_id=self._dataset.id)
-        self.__fields = [field_from_model(model) for model in models]
+        self.__fields = [TextField.from_model(model) for model in models]
```

Comment:
fields returned by the API are only TextField. We should combine fields, vectors, and metadata API results to return all as fields, but I would say to review it and in any case, apply it later.

Reply:
This looks good, but you would leave the naming as is?

Reply:
We should review naming, yes. We'll tackle it later since we have some other places where this is also confusing